Lambdas/GHOAuth/OAuthLambda.swift

```diff
@@ -0,0 +1,194 @@
+import AsyncHTTPClient
+import AWSLambdaRuntime
+import AWSLambdaEvents
+import SotoCore
+import DiscordBM
+import Models
+import JWTKit
+import LambdasShared
+import SharedServices
+import Logging
+import Foundation
+
+@main
+struct GHOAuthHandler: LambdaHandler {
+    typealias Event = APIGatewayV2Request
+    typealias Output = APIGatewayV2Response
+
+    let client: HTTPClient
+    let logger: Logger
+    let secretsRetriever: SecretsRetriever
+    let userService: UserService
+
+    let jsonDecoder = JSONDecoder()
+    let jsonEncoder = JSONEncoder()
+
+    /// We don't do this in the initializer to avoid a possible unnecessary
+    /// `secretsRetriever.getSecret()` call which costs $$$.
+    var discordClient: any DiscordClient {
+        get async throws {
+            let botToken = try await secretsRetriever.getSecret(arnEnvVarKey: "BOT_TOKEN_ARN")
+            return await DefaultDiscordClient(httpClient: client, token: botToken)
+        }
+    }
+
+    init(context: LambdaInitializationContext) async throws {
+        self.client = HTTPClient(eventLoopGroupProvider: .shared(context.eventLoop))
+        self.logger = context.logger
+
+        let awsClient = AWSClient(httpClientProvider: .shared(client))
+        self.secretsRetriever = SecretsRetriever(awsClient: awsClient, logger: logger)
+
+        self.userService = UserService(awsClient, logger)
+    }
+
+    func handle(_ event: APIGatewayV2Request, context: LambdaContext) async -> APIGatewayV2Response {
+        logger.debug("Received event: \(event)")
+
+        guard let code = event.queryStringParameters?["code"] else {
+            return .init(statusCode: .badRequest, body: "Missing code query parameter")
+        }
+
+        let accessToken: String
+
+        do {
+            accessToken = try await getGHAccessToken(code: code)
+        } catch {
+            logger.error("Error getting access token", metadata: [
+                "error": "\(error)"
```

Comment:
```suggestion
                "error": "\(String(reflecting: error))"
```
I've been trying to get into the habit of always logging errors with their debug descriptions (i.e. with `String(reflecting:)`) rather than with their "default" descriptions; many errors provide much more useful info that way. (See also the debacle Fabian and I went through with making PostgresNIO's errors less info-leaking by default.)